OpenSim/Common/Component.cpp

```diff
@@ -157,6 +157,9 @@ void Component::finalizeFromProperties()
 {
     reset();
 
+    // last opportunity to modify Object names based on properties
+    makeObjectNamesConsistentWithProperties();
```

Comment:
It seems to me that we are running this function multiple times per object (note that `componentsFinalizeFromProperties()` descends the ownership tree).

Reply:
Yes, but not all Objects are Components. A couple cases that fail if we don't traverse all objects are use of `Function` and `SpatialTransform`.

Reply:
Ah okay. Well we really should find a way to avoid these repeated calls. Can you have only Model invoke this function in its `extendFinalizeFromProperties()`?